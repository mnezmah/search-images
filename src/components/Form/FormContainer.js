import React, { Component } from 'react'
import { connect } from 'react-redux'
import Form from './Form'
import { getImages } from '../../actions/images'

class FormContainer extends Component {
  state = {
    value: ''
  }

  onSubmit = (e) => {
    e.preventDefault()
    this.props.getImages(this.state.value)
    this.setState({
      value: ''
    })
  }

  onChangeHandler = (e) => {
    this.setState({
      value: e.target.value
    })
  }

  render() {
    return (
      <div>
        <Form value={this.state.value}
          onChangeHandler={this.onChangeHandler}
          onsubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    images: state.images
  }
}

const mapDispatchToProps = {
  getImages
}

export default connect(mapStateToProps, mapDispatchToProps)(FormContainer)