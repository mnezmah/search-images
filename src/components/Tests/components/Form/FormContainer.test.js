import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import FormContainer from '../../../Form/FormContainer'

const mockStore = configureMockStore();
const store = mockStore({});

test('should render FormContainer corectly', () => {
  const wrapper = shallow(
    <Provider store={store}><FormContainer /></Provider>
  )
  expect(wrapper).toMatchSnapshot()
})