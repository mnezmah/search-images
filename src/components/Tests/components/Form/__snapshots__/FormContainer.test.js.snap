// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormContainer corectly 1`] = `ShallowWrapper {}`;
